app/services/waste_exemptions_engine/confirmation_pdf_generator_service.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module WasteExemptionsEngine
+  class ConfirmationPdfGeneratorService < BaseService
+    def run(registration:)
+      @registration = registration
+
+      pdf_generator.pdf
+    end
+
+    private
+
+    def pdf_generator
```

Comment:
Sure it can. But, since the PDF generation service is an external resource to this service, is good practice to keep it in its own method. I suggest this reading: https://www.poodr.com/ it exposes very clearly the facts for such choices :)

Reply:
Leave as submitted for now, and let's add it as another thing we discuss and agree next Tuesday in our dev meeting. You can also give us the TL;DR as life gives me about 3 minutes a day for reading, so it'd be awhile till I get to those facts!

Reply:
Long story short, there are 2 main benefits: 
- As the class grow (if/when) the method name is also documenting the purpose of the object or the logic inside the method. 
- As the code change, and so the PDF generator can change, having the instance created in its own method means that will be easier to update it to a different object name, given that the public interface of the generator object is unchanged.